docs/upgrade-guide.md

```diff
@@ -1,5 +1,86 @@
 # Upgrade Guide
 
+## Upgrading from v4 to v5
+
+Please read this documentation before upgrading your luma.gl's dependency from v4 to v5. In v5 a number of previously deprecated features have been removed and a number of additional deprecations have been made at the same time.
+
+Before upgrading to v5, it is highly recommended to run your application using latest v4 release, and check the console for any deprecated warnings, if there are any replace deprecated API with newer API as listed below.
+
+### Model Class
+
+The `Model` constructor expects a gl context as the first argument.
+
+```
+  // v5
+  Model(gl)
+  Model(gl, {...opts});
+  Model(gl, {program});
+```
+
+Following style construction was deprecated in v4 and is now removed in v5.
+
+```
+  // NOT SUPPORTED
+  Model(gl)
+  Model(gl, {...opts});
+  Model(gl, {program});
+```
+
+### useDevicePixelRatio
+
+`useDevicePixelRatio` is used as a an argument in `AnimationLoop` class constructor and `pickModels` method. It is now deprecated in v5, but still supported with a warning message and will be removed in next major version update. It is recommended to use `useDevicePixels` instead.
+
+### Geometry
+
+`Geometry` class construction with inline attributes was deprecated in v4 and now removed in v5.
+
+```
+// NOT SUPPORTED
+new Geometry({
+  positions: new Float32Array([ ... ]),
+  colors: {
+    size: 4,
+    value: new Float32Array([ ... ])
+  }
+});
+```
+
+All attributes should be grouped inside `attribute` object.
+
+```
+// SUPPORTED
+new Geometry({
+ attributes: {
+   positions: new Float32Array([ ... ]),
+   colors: {
+     size: 4,
+     value: new Float32Array([ ... ])
+   }
+ }
+});
+```
+
+### Removed Features
```

Comment:
Have luma.gl math classes been removed?

Reply:
I removed ` src/deprecated/math-old.js`, are you referring to that?